.circleci/config.yml

```diff
@@ -1,72 +1,79 @@
+references:
+  attach_workspace: &attach_workspace
+    attach_workspace:
+      at: ~/app
+
+defaults: &defaults
+  working_directory: ~/app
+  docker:
+      - image: circleci/node:10-browsers
+
 jobs:
+  build:
+    <<: *defaults
+    steps:
+      - checkout
+      - run:
+          command: 'yarn install'
+          name: 'Install dependencies'
+      - run:
+          command: 'yarn run build:app'
+          name: 'Build app'
+      - persist_to_workspace:
+          root: ~/app
+          paths:
+            - .cache
+            - public
+            - node_modules
   lighthouse:
-    docker:
-      - image: circleci/node:10-browsers
+    <<: *defaults
     steps:
       - checkout
-      - restore_cache:
-          keys:
-            - v1-dependencies-{{ checksum "package.json" }}
-            - v1-dependencies
+      - *attach_workspace
       - run:
           command: npm run test:lighthouse
           name: Lighthouse
-    working_directory: ~/repo
   lint:
-    docker:
-      - image: circleci/node:10
+    <<: *defaults
     steps:
       - checkout
-      - restore_cache:
-          keys:
-            - v1-dependencies-{{ checksum "package.json" }}
-            - v1-dependencies
-      - run: 'yarn install'
-      - save_cache:
-          key: v1-dependencies-{{ checksum "package.json" }}
-          paths:
-            - node_modules
+      - *attach_workspace
+      - run: pwd
+      - run: ls -lha
       - run: 'yarn lint'
-    working_directory: ~/repo
   snapshots:
-    docker:
-      - image: circleci/node:10
+    <<: *defaults
     steps:
       - checkout
-      - restore_cache:
-          keys:
-            - v1-dependencies-{{ checksum "package.json" }}
-            - v1-dependencies
+      - *attach_workspace
       - run:
           command: npm run test:snapshots
           name: Snapshots
-    working_directory: ~/repo
   testcafe:
-    docker:
-      - image: circleci/node:10-browsers
+    <<: *defaults
     steps:
       - checkout
-      - restore_cache:
-          keys:
-            - v1-dependencies-{{ checksum "package.json" }}
-            - v1-dependencies
-      - run: yarn install
-      - save_cache:
-          paths:
-            - node_modules
-          key: v1-dependencies-{{ checksum "package.json" }}
+      - *attach_workspace
       - run:
           command: npm run test:e2e
           name: Testcafe
       - store_artifacts:
-          path: ~/repo/artifacts/screenshots
-    working_directory: ~/repo
+          path: ~/artifacts/screenshots
 version: 2
 workflows:
   build_and_test:
     jobs:
-      - lint
-      - lighthouse
-      - snapshots
-      - testcafe
+      - build
+      - lint:
```

Comment:
I understand adding build to all the commands that had `npm run build` as first part of the command. But why does lint need it if it doesn't need the build to happen to work?

Reply:
Actually we do need this as we use the node modules from build for lint